chia_py_streamable_macro/src/lib.rs

```diff
@@ -154,17 +154,34 @@ pub fn py_streamable_macro(input: proc_macro::TokenStream) -> proc_macro::TokenS
                 <Self as #crate_name::Streamable>::from_bytes(slice).map_err(|e| <#crate_name::chia_error::Error as Into<pyo3::PyErr>>::into(e))
             }
 
+            #[staticmethod]
+            #[pyo3(name = "from_bytes_unchecked")]
+            pub fn py_from_bytes_unchecked(blob: pyo3::buffer::PyBuffer<u8>) -> pyo3::PyResult<Self> {
+                if !blob.is_c_contiguous() {
+                    panic!("from_bytes_unchecked() must be called with a contiguous buffer");
```

Comment:
Maybe `ValueError` is better here?

Reply:
it would make me a bit uneasy to not have a hard failure. I think it's a problem, in `chia-blockchain`, that we swallow serious errors (sometimes assertion failures, and sometimes without logging) and keep going. I suppose a panic here also results in a python exception, but at least one that stands out a bit more and less likely to be swallowed.

Reply:
I'm actually unclear on which circumstances we would get a non-contiguous buffer from python. I imagine both `bytes` and `bytearray` are guaranteed to be contiguous, no?

Reply:
Seems so.

Easiest way I've found to create non-contiguous is as follows:

```
>>> memoryview(b"foobar")[::2].contiguous
False
```